Carrito.js

```diff
@@ -0,0 +1,193 @@
+                        <span class="carrito-precio">$${this.formateador.format(producto.precio)}</span>
+                        <span class="carrito-cantidad">
+                            <button class="btn-decrementar" data-nombre="${producto.nombre}">-</button>
+                            ${producto.cantidad}
+                            <button class="btn-incrementar" data-nombre="${producto.nombre}">+</button>
+                        </span>
+                        <button class="btn-eliminar" data-nombre="${producto.nombre}">Eliminar</button>
+                    </li>
+                `
+            )
+            .join("");
+
+        document.querySelectorAll(".btn-decrementar").forEach((button) => {
+            button.addEventListener("click", () => this.decrementarCantidad(button.dataset.nombre));
+        });
+
+        document.querySelectorAll(".btn-incrementar").forEach((button) => {
+            button.addEventListener("click", () => this.incrementarCantidad(button.dataset.nombre));
+        });
+
+        document.querySelectorAll(".btn-eliminar").forEach((button) => {
+            button.addEventListener("click", () => this.eliminarDelCarrito(button.dataset.nombre));
+        });
+
+        const totalContainer = document.querySelector("#carrito-total span");
+        if (totalContainer) {
+            totalContainer.textContent = this.calcularTotal();
+        }
+    }
+
+    inicializarBotones() {
+        const botonesAgregar = document.querySelectorAll(".add-to-cart-button");
+
+        botonesAgregar.forEach((boton) => {
+            boton.addEventListener("click", () => {
+                const nombreProducto = boton.dataset.nombre;
+                const precioProducto = parseFloat(boton.dataset.precio);
+
+                if (!this.validarProducto(nombreProducto, precioProducto)) {
+                    console.error("Datos inválidos al agregar producto");
+                    return;
+                }
+
+                this.agregarAlCarrito(nombreProducto, precioProducto);
+            });
+        });
+    }
+
+    async cargarProductosAPI() {
+        try {
+            const respuesta = await fetch("https://fakestoreapi.com/products");
+            const productos = await respuesta.json();
+
+            this.productos = productos.map((producto) => ({
+                nombre: producto.title,
+                precio: producto.price,
+                cantidad: 1,
+                imagen: producto.image,
+            }));
+
+            this.guardarCarrito();
+            this.actualizarCarrito();
+            this.mostrarProductosEnHTML();
+        } catch (error) {
+            console.error("Error al cargar productos desde la API:", error);
+        }
+    }
+
+    mostrarProductosEnHTML() {
+        const contenedorProductos = document.getElementById("productos");
+        contenedorProductos.innerHTML = "";
+
+        this.productos.forEach((producto) => {
+            const productoHTML = `
```

Comment:
En lugar de usar innerHTML +=, es más eficiente crear y añadir directamente nuevos elementos al DOM usando createElement y appendChild.
```suggestion
         const productoDiv = document.createElement('div');
         productoDiv.classList.add('producto', 'card');
         productoDiv.innerHTML = `
```